mongodb-agent on OpenShift fails to get current user

When deploying the operator on OpenShift, the mongodb Pods from the StatefulSet fail to start:

```
$ oc logs mongodb-0 -c mongodb-agent
panic: Failed to get current user: user: unknown userid 1001130000
 
goroutine 1 [running]:
com.tengen/cm/util.init.3()
        /data/mci/bd62cd177fa9529ccacb53e13b1bf2a3/mms-automation/build/go-dependencies/src/com.tengen/cm/util/user.go:14 +0xe5
```

--- a/README.md
+++ b/README.md
@@ -107,7 +107,7 @@ If you want to deploy the operator on OpenShift you will have to provide the env
 
 See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the required configuration for a MongoDB ReplicaSet.
 
-See [here](deploy/operator_openshift.yaml) for an example of how to provide the variable for the Operator deployment.
+See [here](deploy/operator_openshift.yaml) for an example of how to configure the Operator deployment.
 
 #### Example
 

--- a/README.md
+++ b/README.md
@@ -106,6 +106,7 @@ If you update `spec.version` to a later version, consider setting `spec.featureC
 If you want to deploy the operator on OpenShift you will have to provide the environment variable `MANAGED_SECURITY_CONTEXT` set to `true` for both the mongodb container, the mongo container and the operator deployment.
 
 See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the variable for a ReplicaSet deployment.
+
 See [here](deploy/operator_openshift.yaml) for an example of how to provide the variable for the Operator deployment.
 
 #### Example

--- a/README.md
+++ b/README.md
@@ -103,7 +103,7 @@ If you update `spec.version` to a later version, consider setting `spec.featureC
 
 ### Deploying on OpenShift
 
-If you want to deploy the operator on OpenShift you will have to provide the environment variable `MANAGED_SECURITY_CONTEXT` set to `true` for both the mongodb container, the mongo container and the operator deployment.
+If you want to deploy the operator on OpenShift you will have to provide the environment variable `MANAGED_SECURITY_CONTEXT` set to `true` for both the mongodb and mongodb agent containers, as well as the operator deployment.
 
 See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the variable for a ReplicaSet deployment.
 

--- a/README.md
+++ b/README.md
@@ -105,7 +105,7 @@ If you update `spec.version` to a later version, consider setting `spec.featureC
 
 If you want to deploy the operator on OpenShift you will have to provide the environment variable `MANAGED_SECURITY_CONTEXT` set to `true` for both the mongodb and mongodb agent containers, as well as the operator deployment.
 
-See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the variable for a ReplicaSet deployment.
+See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the required configuration for a MongoDB ReplicaSet.
 
 See [here](deploy/operator_openshift.yaml) for an example of how to provide the variable for the Operator deployment.
 

--- a/README.md
+++ b/README.md
@@ -101,6 +101,13 @@ You can upgrade the major, minor, and/or feature compatibility versions of your 
 
 If you update `spec.version` to a later version, consider setting `spec.featureCompatibilityVersion` to the current working MongoDB version to give yourself the option to downgrade if necessary. To learn more about feature compatibility, see [`setFeatureCompatibilityVersion`](https://docs.mongodb.com/manual/reference/command/setFeatureCompatibilityVersion/) in the MongoDB Manual.
 
+### Deploying on OpenShift
+
+If you want to deploy the operator on OpenShift you will have to provide the environment variable `MANAGED_SECURITY_CONTEXT` set to `true` for both the mongodb container, the mongo container and the operator deployment.
+
+See [here](deploy/crds/mongodb.com_v1_mongodb_openshift_cr.yaml) for an example of how to provide the variable for a ReplicaSet deployment.
+See [here](deploy/operator_openshift.yaml) for an example of how to provide the variable for the Operator deployment.
+
 #### Example
 
 Consider the following example MongoDB resource definition:

--- a/deploy/operator_openshift.yaml
+++ b/deploy/operator_openshift.yaml
@@ -27,7 +27,7 @@ spec:
             - name: POD_NAME
               valueFrom:
                 fieldRef:
-                  fieldPath: metadata.name-
+                  fieldPath: metadata.name
             - name: MANAGED_SECURITY_CONTEXT
               value: 'true'
             - name: OPERATOR_NAME

--- a/deploy/operator_openshift.yaml
+++ b/deploy/operator_openshift.yaml
@@ -1,0 +1,38 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: mongodb-kubernetes-operator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: mongodb-kubernetes-operator
+  template:
+    metadata:
+      labels:
+        name: mongodb-kubernetes-operator
+    spec:
+      serviceAccountName: mongodb-kubernetes-operator
+      containers:
+        - name: mongodb-kubernetes-operator
+          image: quay.io/mongodb/mongodb-kubernetes-operator:0.2.0
+          command:
+          - mongodb-kubernetes-operator
+          imagePullPolicy: Always
+          env:
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name-
+            - name: MANAGED_SECURITY_CONTEXT
+              value: 'true'
+            - name: OPERATOR_NAME
+              value: "mongodb-kubernetes-operator"
+            - name: AGENT_IMAGE # The MongoDB Agent the operator will deploy to manage MongoDB deployments
+              value: quay.io/mongodb/mongodb-agent:10.15.1.6468-1
+            - name: VERSION_UPGRADE_HOOK_IMAGE
+              value: quay.io/mongodb/mongodb-kubernetes-operator-version-upgrade-post-start-hook:1.0.2

